packages/core/framework/src/http/middlewares/apply-locale.ts

```diff
@@ -0,0 +1,39 @@
+import { normalizeLocale } from "@medusajs/utils"
+import type {
+  MedusaNextFunction,
+  MedusaRequest,
+  MedusaResponse,
+} from "../types"
+
+const CONTENT_LANGUAGE_HEADER = "content-language"
```

Comment:
todo: So, this actually seems to be a response header, not a request header, so it is relatively unusual to be using it like this. Instead we should be using `Accept-Language` and extract the locale from there. The locales in this header typically come with weights, we can use for the selection. That is of cause, if the locale is not singular.

Reply:
Accept-language is not for language selection I believe, but more if no locale is being provided, then we can pick from their and choose the one with the higher weight if available. This header is meant as a Hint for the server if it can find the prefered locale for the user

Reply:
Fine to start with this, and we can reevaluate if need be.